ansible_base/resource_registry/apps.py

```diff
@@ -50,7 +51,20 @@ def initialize_resources(sender, **kwargs):
             else:
                 resource_type = f"{registry.api_config.service_type}.{content.model}"
             defaults = {"externally_managed": resource_config.externally_managed, "name": resource_type}
-            ResourceType.objects.update_or_create(content_type=content, defaults=defaults)
+
+            try:
+                ResourceType.objects.update_or_create(content_type=content, defaults=defaults)
+            except IntegrityError as e:
+                # if previous DAB migrations used the wrong content type id, we need to correct that now
+                # to eliminate integrity errors at the end of the migration process when this function
+                # gets called.
+                if not ResourceType.objects.filter(name=resource_type).exists():
+                    raise e
+                rt = ResourceType.objects.get(name=resource_type)
```

Comment:
I would suggest - take your comment from L58-L60, turn it into a log message, and move it here.

It would be good to include the model, new and old, in that log. These are unique on (app_label, model). So in log, give (app_label, model) of `rt.content_type`, and of `content`.

If I understand correctly, this should be a very rare and notable event. I would like for a log to create a record of it.

Reply:
Okay, the log message will look like this on a galaxy system ... 
```
pulp [None]: ansible_base.resource_registry.apps:WARNING: changing content_type for 'shared.team' from 'group' to 'team'
```